src/Microsoft.DotNet.ServiceFabric.ServiceHost/ServiceHostEventListener.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.ServiceFabric.ServiceHost;
+
+public class ServiceHostEventListener : EventListener
+{
+    private static ServiceHostEventListener _instance;
+    private static IImmutableList<string> _eventSourceNamePrefixes;
+    private static ILoggerFactory _loggerFactory;
+    private static ConcurrentDictionary<string, ILogger> _loggers = new();
+
+    public static ServiceHostEventListener ListenToEventSources(ILoggerFactory loggerFactory, params string[] namePrefixes)
+    {
+        if (_instance is not null)
+        {
+            throw new InvalidOperationException("Cannot call ServiceHostEventListener.ListenToEventSources more than once.");
+        }
+        // These fields must be static because the "OnEventSourceCreated" method gets called inside the base constructor, so instance fields aren't initialized when it runs
```

Comment:
That's such an... odd... design choice.  How is one meant to make a useful EventListener in .NET?  Surely they aren't all forced to be static?

Reply:
I think they generally all have hardcoded lists of sources to listen to.

Reply:
This is even called out in the docs, that both OnEventSourceCreated and OnEventWritten can be called before your .ctor

Reply:
That's... so weird and lame.  It really is a terrible interface then.